specs/rate-limiting.md

```diff
@@ -0,0 +1,43 @@
+# Rate limiting guidance
+
+Using the rate limiting guidance plugin, developers can ensure that they properly handle rate limiting in their applications.
+
+Rate limiting is an API feature that communicates to API consumers how many resources are left in a given time window. When the API consumer reaches the limit, the API will return a `429 Too Many Requests` response. API consumers can use information about rate limiting to avoid reaching the limit and to build applications that can stay under the throttling limit.
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | ---- | -------- | ------ |
+| 1.0 | 2023-03-08 | Initial specifications | @waldekmastykarz |
+
+## Configuration
+
+Setting|Description
+-------|-----------
+`headerLimit`|Name of the `RateLimit-Limit` header returned by the API. Default: `RateLimit-Limit`
+`headerRemaining`|Name of the `RateLimit-Remaining` header returned by the API. Default: `RateLimit-Remaining`
+`headerReset`|Name of the `RateLimit-Reset` header returned by the API. Default: `RateLimit-Reset`
+`headerRetryAfter`|Name of the `Retry-After` header returned by the API. Default: `Retry-After`
+`costPerRequest`|Number of resources consumed by a single request. Default: `2`
+`resetTimeWindowSeconds`|Number of seconds before the resource counter is reset. Default: `60`
+`warningThresholdPercent`|Percentage of remaining resources at which the warning should be displayed. Default: `80`
+`rateLimit`|Number of resources available in the time window. Default: `120`
+`retryAfterSeconds`|Number of seconds to wait before retrying the request. Default: `5`
+
```

Comment:
Can this plugin define its own set of URLs to watch or must it follow the overall set?

Reply:
I assumed that configuring own set of URL is a common base functionality of all plugins which is why I haven't mentioned it here explicitly.